File: frontend/src/services/bedService.js
import api from "./api";

const createBed = async (roomId, bedData) => {
    const response = await api.post(`/rooms/${roomId}/beds`, bedData);
    return response.data;
};

const getBedsByRoom = async (roomId) => {
    const response = await api.get(`/rooms/${roomId}/beds`);
    return response.data;
};

const getBedById = async (bedId) => {
    const response = await api.get(`/beds/${bedId}`);
    return response.data;
};

const updateBed = async (bedId, bedData) => {
    const response = await api.put(`/beds/${bedId}`, bedData);
    return response.data;
};

const deleteBed = async (bedId) => {
    await api.delete(`/beds/${bedId}`);
};

const bedService = {
    createBed,
    getBedsByRoom,
    getBedById,
    updateBed,
    deleteBed,
};

export default bedService;

File: src/main/java/com/nishtha/RoomEase/bed/service/BedService.java
package com.nishtha.RoomEase.bed.service;

import com.nishtha.RoomEase.bed.dto.AddBedRequest;
import com.nishtha.RoomEase.bed.dto.BedResponse;
import com.nishtha.RoomEase.bed.dto.UpdateBedRequest;

import java.util.List;

public interface BedService {

    BedResponse createBed(Long roomId, AddBedRequest request);

    List<BedResponse> getBedsByRoom(Long roomId);

    BedResponse getBedById(Long bedId);

    BedResponse updateBed(Long bedId, UpdateBedRequest request);

    void deleteBed(Long bedId);

}